src/components/dialogs/filters.js

```diff
@@ -290,6 +301,256 @@ export const RangeSelection = ({ initialValue, onChange, titleMessage }) => {
+                    multiple={true}
+                    onChange={(oldVal, newVal) => {
+                        onChange(index, { name, values: newVal });
+                        setValues(newVal);
+                    }}
+                    options={predefinedValues}
+                    renderInput={(props) => (
+                        <TextField
+                            label={<FormattedMessage id="PropertyValues" />}
+                            {...props}
+                        />
+                    )}
+                    renderTags={(val, getTagsProps) =>
+                        val.map((code, index) => (
+                            <Chip
+                                id={'chip_' + code}
+                                size={'small'}
+                                label={code}
+                                {...getTagsProps({ index })}
+                            />
+                        ))
+                    }
+                />
+            </FormControl>
+        </>
+    );
+};
+
+const FreePropertiesOneSide = ({ initialValue, onChange, titleMessage }) => {
+    const numericSuffixRegex = /[0-9]*$/;
+    const numericSuffix = numericSuffixRegex.exec(titleMessage)[0];
+
+    const [fieldProps, setFieldProps] = useState(null);
+    const initialValues = useMemo(() => {
+        if (!initialValue) return [];
+        const ret = Object.entries(initialValue).map(([k, v]) => {
+            return { name: k, values: v };
+        });
+        return ret;
+    }, [initialValue]);
+
+    const onPropertiesArrayChange = useCallback(
+        (arr) => {
+            const obj = !arr
+                ? {}
+                : Object.fromEntries(arr.map((p) => [p.name, p.values]));
+            onChange(obj);
+        },
+        [onChange]
+    );
+
+    const freePropsField = useExpandableCriterium({
+        id: 'freeProp' + numericSuffix,
+        labelAddValue: 'AddFreePropCrit' + numericSuffix,
+        Field: FreePropertyOneSide,
+        fieldProps: fieldProps,
+        initialValues: initialValues,
+        onChange: onPropertiesArrayChange,
+    });
+
+    const fetchPredefinedProperties = () => {
+        return fetchAppsAndUrls().then((res) => {
+            const studyMetadata = res.find(
+                (metadata) => metadata.name === 'Study'
+            );
+            if (!studyMetadata) {
+                return Promise.reject(
+                    'Study entry could not be found in metadatas'
+                );
+            }
+
+            return studyMetadata?.predefinedEquipmentProperties?.substation;
+        });
+    };
+
+    const field = useMemo(() => {
+        return (
+            <>
+                <Grid container item direction="row" spacing={2}>
+                    {freePropsField}
+                </Grid>
+            </>
+        );
+    }, [freePropsField]);
+
+    useEffect(() => {
+        fetchPredefinedProperties().then((p) => setFieldProps(p));
+    }, []);
```

Comment:
I'm afraid this is called many times, while it could be called only once (and if we use properties)

Reply:
There is no catch: I wonder if we have to alert when we cant get predef properties ?